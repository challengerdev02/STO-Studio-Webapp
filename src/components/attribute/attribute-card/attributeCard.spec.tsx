import { cleanup, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { AttributesCard } from './index';

describe('AttributesCard', function () {
  afterEach(cleanup);
  let props = {
    icon: <i className="mc-check-fill mc-1x" />,
    title: 'Lorem ipsum lorem',
    bottomText: 'story line',
    backgroundColor: 'voilet',
  };

  const setup = (extraProps: Record<string, any> = {}) => {
    const utils = render(<AttributesCard {...props} {...extraProps} />);
    return {
      ...utils,
    };
  };

  beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  it('should match with snapshot', () => {
    // @ts-ignore
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it('should test the Attributes Card component paints on screen', () => {
    const { getByTestId } = setup();

    const attributes_card_container = getByTestId('attributes-card-container');
    expect(attributes_card_container).toBeInTheDocument();
  });

  it('should test the Attributes Card component that paints on screen has a title text', () => {
    const { getByTestId } = setup();

    const attributes_card_container_title_text = getByTestId(
      'attributes-card-container-title'
    );
    expect(attributes_card_container_title_text).toBeInTheDocument();
  });

  it('should test the Attributes Card component that paints on screen has a bottom text', () => {
    const { getByTestId } = setup();

    const attributes_card_container_bottom_text = getByTestId(
      'attributes-card-container-bottom-text'
    );
    expect(attributes_card_container_bottom_text).toBeInTheDocument();
  });

  it('should test that every Attributes Card component that paints on screen has an icon', () => {
    const { getByTestId } = setup();

    const card_icon = getByTestId('card-icon');
    expect(card_icon).toBeInTheDocument();
  });
});
