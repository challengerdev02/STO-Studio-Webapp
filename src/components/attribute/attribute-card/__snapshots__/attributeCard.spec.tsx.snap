// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttributesCard should match with snapshot 1`] = `
<div>
  <div
    class="ant-card sc-bdvvtL jYlmOu"
    data-testid="attributes-card-container"
  >
    <div
      class="ant-card-body"
    >
      <div
        class="ant-space ant-space-vertical"
      >
        <div
          class="ant-space-item"
          style="margin-bottom: 8px;"
        >
          <div
            class="ant-space ant-space-horizontal ant-space-align-center"
          >
            <div
              class="ant-space-item"
              style="margin-right: 16px;"
            >
              <div
                class="sc-gsDKAQ hkxojj"
                data-testid="card-icon"
              >
                <i
                  class="mc-check-fill mc-1x"
                />
              </div>
            </div>
            <div
              class="ant-space-item"
            >
              <div
                class="sc-dkPtRN hndXjc"
                data-testid="attributes-card-container-title"
                style="text-transform: uppercase;"
              >
                Lorem ipsum lorem
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-space-item"
        >
          <div
            class="sc-dkPtRN hndXjc"
            data-testid="attributes-card-container-bottom-text"
          >
            story line
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
